adlfs/utils.py

```diff
@@ -1,4 +1,15 @@
-async def filter_blobs(blobs, target_path, delimiter="/"):
+from typing import Optional
+
+
+def match_blob_version(blob, version_id: Optional[str]):
+    return (
+        version_id is None and ("version_id" not in blob or blob["is_current_version"])
```

Comment:
version_id could actually be present but set to `None`, which will make us ignore blobs when `fs.version_aware==True`. So maybe

``suggestion
        version_id is None and (blob.get("version_id") is None or blob["is_current_version"])
```

?

Reply:
Also, there is `blob["version_id"]` a line below, maybe we shouldn't assume that it is always present in `blob` too and do a `get()` instead?

Reply:
should be fixed in bbd9483